src/include/ipc/topology.h

```diff
@@ -62,7 +62,7 @@ enum sof_comp_type {
 struct sof_ipc_comp {
 	struct sof_ipc_cmd_hdr hdr;
 	uint32_t id;
-	enum sof_comp_type type;
+	uint32_t type;
```

Comment:
I'm for this change - IPC structures are one of few locations where fixed-size types must be used, unlike in multiple other locations, where proper C types should be used instead. But just one proposal - would be good to add a comment with `enum X` to these fields to help prevent future mistakes

Reply:
ok, this is fine - its good we can get more coverage with different ISAs, the ARM compiler throwing this up means we should make this change.
One thing, can we inline comment each change to say that the new uint32_t type uses enum X

Reply:
fixed, also added reference to #9378. Should make it easier for ppl to find why the change was done w/o having to look at commit msg.